ext/opentelemetry-exporter-cloud-trace/tests/test_cloud_trace_propagator.py

```diff
@@ -0,0 +1,217 @@
+        )
+
+        header = "{}/{};o=".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), self.valid_span_id
+        )
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "{}/;o=1".format(
+            get_hexadecimal_trace_id(self.valid_trace_id)
+        )
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "/{};o=1".format(self.valid_span_id)
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "{}/{};o={}".format("123", "34", "4")
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_invalid_trace_id(self):
+        header = "{}/{};o={}".format(INVALID_TRACE_ID, self.valid_span_id, 1)
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+        header = "{}/{};o={}".format("0" * 32, self.valid_span_id, 1)
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "0/{};o={}".format(self.valid_span_id, 1)
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "234/{};o={}".format(self.valid_span_id, 1)
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "{}/{};o={}".format(self.too_long_id, self.valid_span_id, 1)
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_invalid_span_id(self):
+        header = "{}/{};o={}".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), INVALID_SPAN_ID, 1
+        )
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "{}/{};o={}".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), "0" * 16, 1
+        )
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "{}/{};o={}".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), "0", 1
+        )
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+        header = "{}/{};o={}".format(
+            get_hexadecimal_trace_id(self.valid_trace_id), self.too_long_id, 1
+        )
+        self.assertEqual(
+            self._extract(header), trace.INVALID_SPAN.get_context()
+        )
+
+    def test_inject_with_no_context(self):
+        output = self._inject()
+        self.assertEqual(output, None)
+
+    def test_inject_with_invalid_context(self):
+        output = self._inject(trace.INVALID_SPAN)
+        self.assertEqual(output, None)
```

Comment:
There is a [specific](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIsNone) function for this case.